File: Course-Registration-System-main/python-(iteration_2)/src/Advisor.py
from Lecturer import Lecturer

class Advisor(Lecturer):
    def __init__(self, lecturer=None, id=None, password=None, name=None, surname=None, title=None, supervisedStudentList=None):
        if lecturer is not None:
            super().__init__(lecturer.getId(), lecturer.getPassword(), lecturer.getName(), lecturer.getSurname(), lecturer.getTitle())
            self.__supervisedStudentList = []
        else:
            super().__init__(id, password, name, surname, title)
            self.__supervisedStudentList = supervisedStudentList if supervisedStudentList is not None else []

    def isSupervisingStudent(self, studentID):
        supervising = False
        studentIdAsInteger = 0
        try:
            studentIdAsInteger = int(studentID)
        except Exception as e:
            return False

        for student in self.__supervisedStudentList:
            if student.getId() == studentIdAsInteger:
                supervising = True

        return supervising

    def getSupervisedStudent(self, studentID):
        studentIdAsInteger = 0
        student = None
        try:
            studentIdAsInteger = int(studentID)
            for supervisedStudent in self.__supervisedStudentList:
                if supervisedStudent.getId() == studentIdAsInteger:
                    student = supervisedStudent
        except Exception as e:
            print()

        return student

    # Getters and Setters for attributes
    def getSupervisedStudentList(self):
        return self.__supervisedStudentList

    def setSupervisedStudentList(self, supervisedStudentList):
        self.__supervisedStudentList = supervisedStudentList

    def getInfo(self):
        # Implementation for viewing general advisor information
        return f"Advisor: {self.getTitle()} {self.getName()} {self.getSurname()}\n"

File: Course-Registration-System-main/java-(iteration_1)/src/Advisor.java
import java.util.ArrayList;

public class Advisor extends Lecturer {
    private ArrayList<Student> supervisedStudentList;

    // Constructor
    public Advisor(Lecturer lecturer) {
        super(lecturer.getId(), lecturer.getPassword(), lecturer.getName(), lecturer.getSurname(), lecturer.getTitle());
        this.supervisedStudentList = new ArrayList<>();

    }

    public Advisor(int id, String password, String name, String surname, String title,
            ArrayList<Student> supervisedStudentList) {
        super(id, password, name, surname, title);
        this.supervisedStudentList = supervisedStudentList;

    }

    public Boolean isSupervisingStudent(String studentID) {
        Boolean supervising = false;
        int studentIdAsInteger = 0;
        try {
            studentIdAsInteger = Integer.parseInt(studentID);
        } catch (Exception e) {
            return false;
        }

        for (int i = 0; i < supervisedStudentList.size(); i++) {
            if (supervisedStudentList.get(i).getId() == studentIdAsInteger)
                supervising = true;
        }
        return supervising;
    }

    public Student getSupervisedStudent(String studentID) {
        int studentIdAsInteger = 0;
        Student student = null;
        try {
            studentIdAsInteger = Integer.parseInt(studentID);
            for (int i = 0; i < supervisedStudentList.size(); i++) {
                if (supervisedStudentList.get(i).getId() == studentIdAsInteger)
                    student = supervisedStudentList.get(i);
            }
        } catch (Exception e) {
            System.err.println();
        }
        return student;
    }

    // Getters and Setters for attributes
    public ArrayList<Student> getSupervisedStudentList() {
        return supervisedStudentList;
    }

    public void setSupervisedStudentList(ArrayList<Student> supervisedStudentList) {
        this.supervisedStudentList = supervisedStudentList;
    }

    @Override
    public String getInfo() {
        // Implementation for viewing general advisor information
        return "Advisor: " + getTitle() + " " + getName() + " " + getSurname();
    }
}
